Sources/StreamChat/Workers/Background/ConnectionRecoveryUpdater.swift

```diff
@@ -9,178 +9,322 @@ import Foundation
 /// was not connected to the web-socket.
 ///
 /// The object listens for `ConnectionStatusUpdated` events
-/// and remembers the `CurrentUserDTO.lastReceivedEventDate` when status becomes `connecting`.
+/// and remembers the `CurrentUserDTO.lastSyncedAt` when status becomes `connecting`.
 ///
 /// When the status becomes `connected` the `/sync` endpoint is called
-/// with `lastReceivedEventDate` and `cids` of watched channels.
+/// with `lastSyncedAt` and `cids` of watched channels.
 ///
-/// We remember `lastReceivedEventDate` when state becomes `connecting` to catch the last event date
-/// before the `HealthCheck` override the `lastReceivedEventDate` with the recent date.
+/// We remember `lastSyncedAt` when state becomes `connecting` to catch the last event date
+/// before the `HealthCheck` override the `lastSyncedAt` with the recent date.
 ///
-class ConnectionRecoveryUpdater: EventWorker {
+class ConnectionRecoveryUpdater {
     // MARK: - Properties
     
-    private var connectionObserver: EventObserver?
-    private let databaseCleanupUpdater: DatabaseCleanupUpdater
-    @Atomic private var lastSyncedAt: Date?
-    private let useSyncEndpoint: Bool
+    private unowned var client: ChatClient
+    
+    @Atomic private var _activeChannelListRefs = NSHashTable<AnyObject>.weakObjects()
+    var activeChannelListControllers: [ChatChannelListController] {
+        _activeChannelListRefs.allObjects.compactMap { $0 as? ChatChannelListController }
+    }
+    
+    @Atomic private var _activeChannelRefs = NSHashTable<AnyObject>.weakObjects()
+    var activeChannelControllers: [ChatChannelController] {
+        _activeChannelRefs.allObjects.compactMap { $0 as? ChatChannelController }
+    }
```

Comment:
Will probably hide controllers under some protocol to define a single interface for recoverable components